packages/react/context-menu/src/ContextMenu.tsx

```diff
@@ -183,19 +195,33 @@ type ContextMenuRootContentPrimitive = Polymorphic.ForwardRefComponent<
 >;
 
 const ContextMenuRootContent = React.forwardRef((props, forwardedRef) => {
-  const { disableOutsidePointerEvents = true, ...contentProps } = props;
   const context = useContextMenuContext(CONTENT_NAME);
+  const shouldPreventFocusRef = React.useRef(false);
```

Comment:
Haha, we literally changed them all the other day because I thought the opposite of what you're describing here. The biggest issue I have is that most of the time (in these at least) it's nearly impossible to name that thing with something actually accurate to "something that occured" because it encapsulates way too many cases. So you end up with a `hasInteractOutsideRef` which doesn't really mean that at all because we've grouped a bunch of different conditions inside it.

I rather in these cases prefer to do the commands thing because where the things happen we see what we're trying to do based on them.

Happy to re-consider if we find good names, but honestly there are some cases that will be impossible.

Reply:
> because we've grouped a bunch of different conditions inside it.

What are the conditions that are grouped inside it? As far as I can see they are if they interact outside and:

- it's not modal
- user hasn't prevented it

Both of these conditions when true mean that the user **hasn't** interacted outside because, they can't when it is modal or prevented.

But like I say, I don't care what we name it as long as it describes **what occurred** and not what it's going to control elsewhere because you lose context when reading the code. 

Everything needs code comments to describe the acceptance criteria rather than it just being self-documented.

Reply:
The worst offender that prompted it all for me was this one:

https://github.com/radix-ui/primitives/blob/4bb5024a2038df264a0bb6dfeb44819ff7776fc1/packages/react/dropdown-menu/src/DropdownMenu.tsx#L243-L253

Before this we ended up with something like:

```tsx
isInteractOutsideRef.current = !isModalLeftClick;
```

Something like that, and I just couldn't get my head around it, I find the new form much easier to understand.
But again, I'm not against doing everything back the other way if we can find a way that reads well but I haven't seen anything convincing yet.

Reply:
In that example I think it landed there because of the modal condition around the assignment, making the naming not technically accurate.

If we rejig it like this do you think it's clearer?

```jsx
onCloseAutoFocus={(event) => {
  props.onCloseAutoFocus?.(event);
  
  if (!event.defaultPrevented && !context.modal && hasInteractedOutsideRef.current) {
    event.preventDefault();
  }

  hasInteractedOutsideRef.current = false;
}}
onInteractOutside={(event) => {
  props.onInteractOutside?.(event);

  if (!event.defaultPrevented) hasInteractedOutsideRef.current = true;
}}
```

Reply:
Yeah that's exactly the point I made before your comment, you might have missed it @andy-hook.

Reply:
Oh snap I missed this entire thread in my stale browser window 🙈

Reply:
I think we're getting too caught up in the conditions. Just because I clicked outside doesn't mean I interacted outside. There are many variables that can prevent that "interaction" and they don't need to be part of the name, they are what determine its truthiness. 

An outside interaction only occurs if they haven't prevented it or if it's non-modal.

```jsx
const hasInteractedOutside = clickedOustide && !modal && !userPrevented;

// If whatever we consider an outside interaction **hasn't occurred* then prevent default
if (!event.defaultPrevented && !hasInteractedOutside) {
    event.preventDefault();
}
```

In the case of the right click stuff I would still just use `hasInteractedOutside` because that is part of what we consider an outside interaction in that case.

Reply:
To compare... 

```jsx
onClick={() => {
	if (!event.defaultPrevented) {
		hasClickedRef.current = true;
	}
}}
``

Are we implying this should actually be named `hasClickedAndUserDidntPreventItRef.current`? 

The latter part is irrelevant in my mind, it's what determines its value.

Reply:
I personally find it hard to envision anything without seeing an example, so I think at this point if we think we want to change things, I need to see the example re-written in the new way to see if I understand it.

Reply:
Sure, the original way @andy-hook wrote it in the PR is the way I am also describing which is quite telling that he did it that way off the bat too.

These changes got merged before I could comment on this but putting it back to something similar to how Andy originally did it gets my vote 👍

Reply:
I think writing it this way makes it a lot more prone to errors or missing a case, etc.
I like the explicitness of the cases where I can step through and understand why we do it.
I find it very confusing and couldn't get my head around it when I was reviewing it, but I'll let you guys settle things then.